### PR TITLE
PLAT-78102: Fixed Measurable to measure after initial render

### PR DESCRIPTION
**Pull request by Djspaceg**
### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [x] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed

* [ ] This is an API breaking change

### Issue Resolved / Feature Added
Occasionally on initial page-load, when Measurable fires, it is actually firing too soon, so it reports the wrong value. This is in-turn fed into Header, which sets the wrong padding, pushing the text content off of the screen.

### Resolution
Refactored Measurable to use a Layout Effect to catch updates after initial render.

This implementation may be "rough", so it's probably ripe for optimization. I'd like to have this only issue a new measurement prop **if and only if** the measurement values have actually changed, but am not sure how to do this. Hoping someone can help brainstorm with me about the "right" way to do this.

### Additional Considerations
This depends on #2251 and #2249.

### CODE REVIEW

File: packages/ui/Measurable/Measurable.js [hunk 1]
```diff
@@ -37,10 +37,43 @@ const defaultConfig = {
...
+	useLayoutEffect(
+		() => {
+			if (!ref.current) {
+				return;
+			}
+
+			measureMe();
+
+			if (typeof ResizeObserver === 'function') {
+				let resizeObserver = new ResizeObserver(() => measureMe());
```
[the remark this answers is not in the document]
Comment (Djspaceg) [reply packages/ui/Measurable/Measurable.js]: That seems much smarter. I got into the weeds on this implementation and was hoping some kind soul like yourself would help me optimize this. Wanna just use what you've suggested, try out some possibly simpler code, or pair up on ways to improve this?

File: packages/ui/Measurable/Measurable.js [hunk 2]
```diff
@@ -37,10 +37,43 @@ const defaultConfig = {
...
+		[ref]
+	);
+
+	useLayoutEffect(
+		() => {
+			if (!ref.current) {
+				return;
+			}
+
+			measureMe();
```
[the remark this answers is not in the document]
Comment (Djspaceg) [reply packages/ui/Measurable/Measurable.js]: I'd love for this whole function to only issue a new set of `measurement` props *if and only if* the values of the measurement object have changed. Think that's possible? There's some built-in stuff in hooks, but it didn't seem elaborate enough to support this, when I looked at it.

**Comment by Djspaceg** [comment]
If the alignment is off for compact, that's unintentional, and I can tweak that to be better. Could have been due to some later Header changes that didn't make it back to Panels. I'll look into that. Thanks.

**Comment by Djspaceg** [comment]
Looked like there was an old unnecessary rule setting the headerComponents' margin-bottom. It has been successfully removed.

**Comment by webOS101** [comment]
Unfortunately, this does not appear to have fixed the issue.  `ActivityPanels` still error with this message:

```
Uncaught TypeError: Object(...) is not a function
    at useMeasurable (Measurable.js:38)
    at MeasurableHoc (Measurable.js:113)
    at mountIndeterminateComponent (react-dom.development.js:14811)
    at beginWork (react-dom.development.js:15316)
    at performUnitOfWork (react-dom.development.js:18150)
    at workLoop (react-dom.development.js:18190)
    at HTMLUnknownElement.callCallback (react-dom.development.js:149)
    at Object.invokeGuardedCallbackDev (react-dom.development.js:199)
    at invokeGuardedCallback (react-dom.development.js:256)
    at replayUnitOfWork (react-dom.development.js:17437)
```

**Pull request merged**